tutorials/sphinx-tutorials/multiagent_competitive_ddpg.py

```diff
@@ -0,0 +1,992 @@
+        x=iteration_when_stop_training_evaders,
+        label="Agent (evader) stop training",
+        color="orange",
+    )
+    axs[i].legend()
+axs[-1].set_xlabel("Training iterations")
+plt.show()
+
+######################################################################
+# Render
+# ------
+#
+# *Rendering instruction are for VMAS*, aka when running with ``use_vmas=True``.
+#
+# If you are running this in a machine with GUI, you can render the trained policy by running:
+#
+# .. code-block:: python
+#
+#    from torchrl.envs.utils import ExplorationType, set_exploration_type
+#    with torch.no_grad(), set_exploration_type(ExplorationType.MODE):
+#       env.rollout(
+#           max_steps=max_steps,
+#           policy=agents_exploration_policy,
+#           callback=lambda env, _: env.render(),
+#           auto_cast_to_device=True,
+#           break_when_any_done=False,
+#       )
+#
+# If you are running this in Google Colab, you can render the trained policy by running:
+#
+# .. code-block:: bash
+#
+#    !sudo apt-get update
+#    !sudo apt-get install python3-opengl xvfb
+#    !pip install pyvirtualdisplay
+#
+# .. code-block:: python
+#
+#    import pyvirtualdisplay
+#    display = pyvirtualdisplay.Display(visible=False, size=(1400, 900))
+#    display.start()
+#    from PIL import Image
+#    from torchrl.envs.utils import ExplorationType, set_exploration_type
+#
+#    def rendering_callback(env, td):
+#        env.frames.append(Image.fromarray(env.render(mode="rgb_array")))
+#    env.frames = []
+#    scenario_name = "simple_tag"
+#    with torch.no_grad(), set_exploration_type(ExplorationType.MODE):
+#       env.rollout(
+#           max_steps=max_steps,
+#           policy=agents_exploration_policy,
+#           callback=rendering_callback,
+#           auto_cast_to_device=True,
+#           break_when_any_done=False,
+#       )
+#    env.frames[0].save(
+#        f"{scenario_name}.gif",
+#        save_all=True,
+#        append_images=env.frames[1:],
+#       duration=3,
+#       loop=0,
+#    )
+#
+#    from IPython.display import Image
+#    Image(open(f"{scenario_name}.gif", "rb").read())
+#
+
+
+######################################################################
+# Conclusion and next steps
+# -------------------------
+#
+# In this tutorial, we have seen:
+#
+# - How to create a competitive multi-group multi-agent environment in TorchRL, how its specs work, and how it integrates with the library;
+# - How to create multi-agent network architectures in TorchRL for multiple groups;
+# - How we can use :class:`tensordict.TensorDict` to carry multi-agent multi-group data;
+# - How we can tie all the library components (collectors, modules, replay buffers, and losses) in a multi-agent multi-group MADDPG/IDDPG training loop.
+#
+# Now that you are proficient with multi-agent DDPG, you can check out all
+# `TorchRL multi-agent examples <https://github.com/pytorch/rl/tree/main/sota-implementations/multiagent>`__.
```

Comment:
let's avoid all links to code if possible

Reply:
I want to reference the sota-implementations folder. How else can I do it?

Reply:
Not sure but it's dangerous because we don't test the links so if we move things from a tiny bit we'll have to manually go through the entire repo looking for this link.

Reply:
Still if there is no alternative I would prefer to keep the reference as I think the users miss out if they are not aware of these code examples